### PR TITLE
MER-312 Bugfixes

### PR DESCRIPTION
**Pull request by gastonabella**
This PR addresses the following bugs:
- [[MER-490]](https://eliterate.atlassian.net/browse/MER-490) Clear flash messages
- [[MER-491]](https://eliterate.atlassian.net/browse/MER-491) Fix typo
- [[MER-492]](https://eliterate.atlassian.net/browse/MER-492) Community admins can delete other community admins
- [[MER-494]](https://eliterate.atlassian.net/browse/MER-494) Community not created when Name is too long
- [[MER-496]](https://eliterate.atlassian.net/browse/MER-496) Community not created when Key Contact is too long
- [[MER-497]](https://eliterate.atlassian.net/browse/MER-497) Can create a community with a duplicate name with surrounding whitespaces.
- [[MER-498]](https://eliterate.atlassian.net/browse/MER-498) Can only add community admins one at a time

### CODE REVIEW

File: lib/oli_web/live/community_live/new_view.ex [hunk 1]
```diff
@@ -37,7 +37,10 @@ defmodule OliWeb.CommunityLive.NewView do
   end
 
   def handle_event("save", %{"community" => params}, socket) do
-    case Groups.create_community(params) do
+    params
```
Comment (mcass19) [inline lib/oli_web/live/community_live/new_view.ex]: NIT: do it in the same way as below? I personally don't like to pipe into case statements

File: lib/oli/groups.ex [hunk 2]
```diff
@@ -205,6 +205,40 @@ defmodule Oli.Groups do
...
+          {:ok, community_account} ->
+            {[community_account | created], errors}
+
+          {:error, error} ->
+            {created, [error | errors]}
+        end
+      end)
+
+    case errors do
+      [error | _errors] ->
```
Comment (mcass19) [inline lib/oli/groups.ex:234]: NIT: why not pattern match on empty and not empty list?

File: lib/oli_web/live/community_live/show_view.ex [hunk 3]
```diff
@@ -219,28 +219,35 @@ defmodule OliWeb.CommunityLive.ShowView do
...
-      {:error, error} ->
-        message =
-          case error do
-            :author_not_found ->
-              "Community admin couldn't be added. Author does not exist."
+        {:error, error} ->
+          message =
+            case error do
+              :author_not_found ->
+                "Community admin couldn't be added. Author does not exist."
```
Comment (mcass19) [inline lib/oli_web/live/community_live/show_view.ex]: We can have the case where more than one author fail, right? Maybe we can tweak a little bit the message to take this into account?

File: lib/oli_web/live/community_live/show_view.ex [hunk 4]
```diff
@@ -219,28 +219,35 @@ defmodule OliWeb.CommunityLive.ShowView do
...
-            :user_not_found ->
-              "Community member couldn't be added. User does not exist."
+              :user_not_found ->
+                "Community member couldn't be added. User does not exist."
 
-            %Ecto.Changeset{} ->
-              "Community user couldn't be added. It is already associated to the community or an unexpected error occurred."
-          end
+              %Ecto.Changeset{} ->
+                "Community user couldn't be added. It is already associated to the community or an unexpected error occurred."
```
Comment (mcass19) [inline lib/oli_web/live/community_live/show_view.ex]: NIT: `#{user_type}` instead of just `user`?

**Review by mcass19 — COMMENTED** [review]
🚀

File: lib/oli_web/live/community_live/form.ex [hunk 5]
```diff
@@ -15,7 +15,7 @@ defmodule OliWeb.CommunityLive.Form do
         {#if @display_labels}
           <Label class="control-label">Community Name</Label>
         {/if}
-        <TextInput class="form-control" opts={placeholder: "Name"}/>
+        <TextInput class="form-control" opts={placeholder: "Name", maxlength: "256"}/>
```
Comment (jrissler) [inline lib/oli_web/live/community_live/form.ex:18]: Is the field UTF-8mb4 (if so this would be 191 instead of 256)?

**Review by jrissler — APPROVED** [review]

File: lib/oli_web/live/community_live/form.ex (hunk 5 above)
Comment (gastonabella) [reply lib/oli_web/live/community_live/form.ex:18]: We're using UTF8 so we're ok, right?

File: lib/oli_web/live/community_live/show_view.ex (hunk 3 above)
Comment (gastonabella) [reply lib/oli_web/live/community_live/show_view.ex]: FFI: discussed this via slack and decided to collapse all possible error types into one generic error message. This feature will be improved in a future story

**Pull request closed without merging**